### PR TITLE
feat: add support for storer TLS auth

### PR DESCRIPTION
**Pull request by ajpauwels**
- For convenience, the client now generate a storer TLS key and
  TLS certificate which can be copied directly to a local storage
  instance
- The client configures Redact storers to use the appropriate
  config-defined and file-system provided client-TLS certs for
  communication with Redact storers
- Symmetric key name for encryption was updated
- All requests to the storer are now secured via TLS

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ajpauwels**
You can assign the PR to them by writing `/assign @ajpauwels` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
@ajpauwels: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 464c25a3f4e6d37d719ff1b18a820dda3b798885 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/bs-redact-client-pr-40-unit-test-hvr86) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main.rs [hunk 1]
```diff
@@ -208,6 +230,110 @@ async fn main() {
...
+                let storer_tls_key_pkcs8 =
+                    PrivateKeyInfo::new(storer_key.algorithm_identifier(), &storer_tls_key_bytes);
+                let mut pkcs8_file = File::create("pkcs8.pem").unwrap();
+                pkcs8_file
+                    .write_all((*storer_tls_key_pkcs8.to_pem()).as_bytes())
+                    .unwrap();
+            }
+            _ => Err(e).unwrap(),
+        }
+    }
```
Comment (howhelan) [inline src/main.rs:302]: move to redact-storage

File: src/main.rs [hunk 2]
```diff
@@ -240,7 +366,7 @@ async fn main() {
             storer_shared.clone(),
             relayer.clone(),
         ))
-        .with(secure_cors.clone());
+        .with(unsecure_cors.clone());
```
Comment (howhelan) [inline src/main.rs:369]: undo

File: src/main.rs (hunk 1 above)
Comment (ajpauwels) [reply src/main.rs]: moved to redact-store

File: src/main.rs (hunk 2 above)
Comment (ajpauwels) [reply src/main.rs]: done

**Pull request merged**